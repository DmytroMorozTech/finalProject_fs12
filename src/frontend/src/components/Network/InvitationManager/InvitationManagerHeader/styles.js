import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  invitationManagerHeader: {
    paddingTop: theme.spacing(2),
    paddingLeft: theme.spacing(4)
  },

  title: {
    fontWeight: theme.typography.fontWeightRegular
  },

  headerItems: {
    display: 'flex',
    marginTop: theme.spacing(2)
  },

  headerItem: {
    marginTop: theme.spacing(3),
    marginRight: theme.spacing(6),
    paddingBottom: theme.spacing(1),
    cursor: 'pointer',
    textDecoration: 'none',
    color: theme.palette.grey[700]
  },

  active: {
    color: theme.palette.success.main,
    borderBottom: theme.border.success
  }
}))
