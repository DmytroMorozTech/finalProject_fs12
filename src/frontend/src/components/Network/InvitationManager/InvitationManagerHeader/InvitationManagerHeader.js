import styles from './styles'
import { Typography } from '@material-ui/core'
import { NavLink } from 'react-router-dom'

function InvitationManagerHeader (props) {
  const {numberOfInvReceived, numberOfInvSent} = props

  const classes = styles()

  return (
    <div className={classes.invitationManagerHeader}>
      <div>
        <Typography variant="h3" className={classes.title}>
          Manage invitations
        </Typography>
        <div className={classes.headerItems}>
          <NavLink to='/network/invitation_manager/received' className={classes.headerItem} activeClassName={classes.active}>
            <Typography variant='h5' color='inherit'>
            Received ({numberOfInvReceived})
            </Typography>
          </NavLink>
          <NavLink to='/network/invitation_manager/sent' className={classes.headerItem} activeClassName={classes.active}>
            <Typography variant='h5' color='inherit'>
            Sent ({numberOfInvSent})
            </Typography>
          </NavLink>
        </div>
      </div>
    </div>
  )
}

export default InvitationManagerHeader
