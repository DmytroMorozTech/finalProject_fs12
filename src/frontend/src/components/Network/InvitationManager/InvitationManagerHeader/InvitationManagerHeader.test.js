import { shallow } from 'enzyme'
import React from 'react'
import InvitationManagerHeader from './InvitationManagerHeader'

jest.mock('./styles', () => () => ({}))
jest.mock('react-router-dom', () => ({
  NavLink: 'a'
}))

const setUp = (props) => shallow(<InvitationManagerHeader {...props}/>)

describe('should render InvitationManagerHeader component', () => {
  let component

  let numberOfInvReceived = 1
  let numberOfInvSent = 2

  beforeEach(() => {
    component = setUp(
      numberOfInvReceived,
      numberOfInvSent
    )
  })

  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
