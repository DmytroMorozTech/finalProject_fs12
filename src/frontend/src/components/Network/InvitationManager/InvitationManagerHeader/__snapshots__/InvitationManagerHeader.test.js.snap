// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InvitationManagerHeader component to match snapshot 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="h3"
    >
      Manage invitations
    </WithStyles(ForwardRef(Typography))>
    <div>
      <a
        to="/network/invitation_manager/received"
      >
        <WithStyles(ForwardRef(Typography))
          color="inherit"
          variant="h5"
        >
          Received (
          )
        </WithStyles(ForwardRef(Typography))>
      </a>
      <a
        to="/network/invitation_manager/sent"
      >
        <WithStyles(ForwardRef(Typography))
          color="inherit"
          variant="h5"
        >
          Sent (
          )
        </WithStyles(ForwardRef(Typography))>
      </a>
    </div>
  </div>
</div>
`;
